pkg/drivers/kic/oci/network.go

```diff
@@ -69,14 +69,49 @@ func digDNS(ociBin, containerName, dns string) (net.IP, error) {
 	return ip, nil
 }
 
+// gatewayIP inspects oci container to find a gateway IP string
+func gatewayIP(ociBin, containerName string) (string, error) {
+	rr, err := runCmd(exec.Command(ociBin, "container", "inspect", "--format", "{{.NetworkSettings.Gateway}}", containerName))
+	if err != nil {
+		return "", errors.Wrapf(err, "inspect gateway")
+	}
+	if gatewayIP := strings.TrimSpace(rr.Stdout.String()); gatewayIP != "" {
+		return gatewayIP, nil
+	}
+
+	// https://github.com/kubernetes/minikube/issues/11293
+	// need to check nested network
+	// check .NetworkSettings.Networks["cluster-name"].Gateway and then
+	// .NetworkSettings.Networks["bridge"|"podman"].Gateway
+	for _, network := range []string{containerName, defaultBridgeName(ociBin)} {
```

Comment:
in this for loop,  we have two things to iterate 
containerName and defaultBridgeName(ociBin)
will the second iteration ever happen ? (since it is returning error if there is nothing for the first one?

Reply:
It returns an error, if `err != nil`. if `err` is not `nil' `gatewayIP` still can be empty

Reply:
so in what case the Second iteration will run ?
```
for _, network := range []string{containerName, defaultBridgeName(ociBin)}
```
first iteration  - containerName
second iteration - defaultBridgeName

as far I see as soon on the first call we Return error or Nil so it will not get to the second iteration, will it ?

Reply:
non-nil err on line 74 means `docker container inspect` is broken. if we just don't  have the gateway ip for the specified network, err is nil. Both `If`s on lines 75 and 78 are false and we go to the second interation

Reply:
ok so lets add a comment here something like so next person reading this code dont have to decrypt it

// first trying to get .... then if there is no err and also the gateway IP is "" we will try the default brdige name....

Reply:
done!